tools/wave/testing/tests_manager.py

```diff
@@ -140,9 +140,32 @@ def compare(tests_manager, test_a, test_b):
                 return -1
             return 1
 
-        sorted_tests.sort(cmp=lambda test_a,
-                          test_b: compare(self, test_a, test_b))
-        return sorted_tests
+        # Helper to adapt "compare" to Python 3 key function.
+        def cmp_to_key(mycmp, test_manager):
+            class K:
```

Comment:
This kind of inner class is probably OK here, but comes with some risk since they don't pickle, so they aren't usable across multiprocessing processes. It might be better to make it a top-level class with `mycmp` and `test_manager` as members, and just create an instance with each call to the key function like `key=lambda x:CmpWrapper(test_manager, mycmp, x)`. I suppose an even better solution would replace the compare function with something that provides an actual key, so we avoid some overhead here, but I see that would be more work.